src/v2/Artsy/Relay/middleware/cache/cacheMiddleware.ts

```diff
@@ -0,0 +1,93 @@
+import { QueryResponseCache } from "relay-runtime"
+import createLogger from "v2/Utils/logger"
+import { Cache, CacheConfig } from "./Cache"
+import { isFunction } from "lodash"
+
+const logger = createLogger("v2/Artsy/middleware/cache/cacheMiddleware")
+
+interface CacheMiddlewareOpts extends CacheConfig {
+  onInit?: (cache: QueryResponseCache) => any
+  allowMutations?: boolean
+  allowFormData?: boolean
+  clearOnMutation?: boolean
+  cacheErrors?: boolean
+  disableServerSideCache?: boolean
+}
+
+/**
+ * Adapted from `react-relay-network-modern`.
+ * @see https://github.com/relay-tools/react-relay-network-modern/tree/master/src/middlewares
+ */
+export function cacheMiddleware(opts?: CacheMiddlewareOpts) {
+  const {
+    size,
+    ttl,
+    onInit,
+    allowMutations,
+    allowFormData,
+    clearOnMutation,
+    cacheErrors,
+    disableServerSideCache = true,
+  } = opts || {}
+
+  const cache = new Cache({
+    size: size || 100, // 100 requests
+    ttl: ttl || 15 * 60 * 1000, // 15 minutes
+    disableServerSideCache,
+  })
+
+  if (isFunction(onInit)) {
+    onInit(cache)
+  }
+
+  return next => async req => {
+    if (req.isMutation()) {
+      if (clearOnMutation) {
+        await cache.clear()
+      }
+      if (!allowMutations) {
+        return next(req)
+      }
+    }
+
+    if (req.isFormData() && !allowFormData) {
+      return next(req)
+    }
+
+    if (disableServerSideCache || (req.cacheConfig && req.cacheConfig.force)) {
+      const queryId = req.getID()
+      const variables = req.getVariables()
+      const res = await next(req)
+
+      if (!res.errors || (res.errors && cacheErrors)) {
```

Comment:
Wondering, in this branch we're skipping the cache, so do we not need this stuff? ie- it should just return res

Reply:
This relates to the `cacheErrors` setting and was copied over from the lib. No need to mess with it I don't think.

Reply:
Yea, but the conditional code branch we're in has caching disabled entirely here (`if disableServerSideCache...`), so thinking that should just skip this errors thing too.

Reply:
Ahh yeah, makes sense 👍 seems weird to cache errors anyway, more just thinking minimal possible changes. Down with whatever!